src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+    * @param updatedUsers The user(s) that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsFrozen status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isFrozenUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsFrozen = new Map<Id, Boolean>();
+
+		// Retrieve the UserLogin records for the updatedUsers
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			Map<Id, Boolean> userLoginUserIdToIsFrozen = retrieveUserLogins(updatedUsers);
+			
+			// Populate the Map with UserId and IsFrozen
+			for(User currentUser : updatedUsers) {
+				userIdToIsFrozen.put(currentUser.Id, userLoginUserIdToIsFrozen.get(currentUser.Id));
+			}
+		}
+
+		return userIdToIsFrozen;
+	}
+	
+
+	/*********************************************************************************************************
+    * @description Handles User changes to IsActive/IsFrozen status by updating the Error Notification Recipients, logging an Error__c, and notifying the org's System Administrators.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	@future
+	public static void applySysAdminIsActiveIsFrozenStatusChanges(Set<Id> sysAdminUserIds) {
+		// Sets Error Notification recipient to All Sys Admin
+		setErrorNotificationToAllSysAdmins(sysAdminUserIds);
+
+		// Logs an Error and Emails the Sys Admins
+		handleScheduledJobsSubmittedByInActiveIsFrozenSysAdmin(sysAdminUserIds);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Sets the NPSP Settings Error Notification Recipients to All Sys Admins.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	public static void setErrorNotificationToAllSysAdmins(Set<Id> sysAdminUserIds) {
+		Error_Settings__c errorSettings = UTIL_CustomSettingsFacade.getErrorSettings();
+		
+		if(errorSettings.Error_Notifications_To__c.startsWith(OBJECT_ID_PREFIX_USER)) {
+			// Check if one of the sysAdminUserIds is the Error Notification recipient
+			for(Id currentSysAdminId: sysAdminUserIds) {
+				if(errorSettings.Error_Notifications_To__c == currentSysAdminId) {
+					// Set Error Notification recipient to All Sys Admin
+					errorSettings.Error_Notifications_To__c = ERROR_NOTIFICATION_RECIPIENT_ALL_SYS_ADMINS;
+					Database.update(errorSettings, true);
+					break;
+				}
+			}
+		}
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if a User is the owner of the Scheduled Jobs.
```

Comment:
description should also give a hint as to what it does in this case.

Reply:
Updated the description. Thanks!